### PR TITLE
feat: one of the thing → one of the things

### LINKED ISSUE #692: feat: Flag "one of the" used with a singular object
## Description

This is pretty common among non-native English speakers.
Instead of "one of the things" they write "one of the thing".

## Resources

If there are any resources online we can reference that described grammatical rules, let us know.

## Examples

- **One of the** neat **trick** with AVX-512 is that given a mask → **One of the** neat **tricks** with ...

### PR DESCRIPTION
**Pull request by hippietrail**
# Issues 
Resolves #692 

# Description

This is a pretty common mistake I come across on the internet all the time.

"One of the thing" instead of "one of the things", etc.

I think mainly nonnative speakers do it, but maybe it's spread to younger generation native speakers too?

# How Has This Been Tested?

Unit tests made from all the example sentences in the GitHub issue.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests to cover my changes

### CODE REVIEW

File: harper-core/src/linting/one_of_the_singular.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+    linting::{ExprLinter, LintKind, Suggestion, expr_linter::Chunk},
+    spell::Dictionary,
+};
+
+pub struct OneOfTheSingular<D: Dictionary + 'static> {
+    expr: Box<dyn Expr>,
+    dict: D,
+}
+
+pub trait SeqExprExt {
```
Comment (elijah-potter) [inline harper-core/src/linting/one_of_the_singular.rs:13]: This should be made public or merged with the main `SequenceExpr` implementation.

File: harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml [hunk 2]
```diff
@@ -579,6 +579,14 @@ Suggest:
 
 
 
+Lint:    Usage (127 priority)
+Message: |
+     271 | As she said this she looked down at her hands, and was surprised to see that she
+     272 | had put on one of the Rabbit’s little white kid gloves while she was talking.
```
Comment (elijah-potter) [inline harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml]: This is wrong.

**Review by elijah-potter — CHANGES_REQUESTED** [review]
This looks mostly ready. I would like to see a small refactor and that edge case addressed.

File: harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml (hunk 2 above)
Comment (hippietrail) [reply harper-core/tests/text/linters/Alice's Adventures in Wonderland.snap.yml]: > This is wrong.

Oh yeah that was weird. I didn't look into it yet...
